src/main/java/com/mojang/brigadier/tree/CommandNode.java

```diff
@@ -89,6 +89,14 @@ public void addChild(final CommandNode<S> node) {
         }
     }
 
+	public void removeChild(final CommandNode<S> node) {
```

Comment:
Nodes are uniquely identified by their name.
Therefore, and because it is the only information you actually use from the node, the parameter should probably be the 'nodeName' and not the node itself.

Especially since when using the intended Argument-builder syntax to declare a command you do not have a reference to the command node, but you do know the name.

Reply:
Okay, thats a good point